x/concentrated-liquidity/incentives.go

```diff
@@ -500,3 +500,70 @@ func (k Keeper) collectIncentives(ctx sdk.Context, poolId uint64, owner sdk.AccA
 	}
 	return collectedIncentives, nil
 }
+
+// createIncentive creates an incentive record in state for the given pool
+func (k Keeper) createIncentive(ctx sdk.Context, poolId uint64, sender sdk.AccAddress, incentiveDenom string, incentiveAmount sdk.Int, emissionRate sdk.Dec, startTime time.Time, minUptime time.Duration) (types.IncentiveRecord, error) {
+	pool, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return types.IncentiveRecord{}, err
+	}
+
+	// Ensure incentive amount is nonzero and nonnegative
+	if !incentiveAmount.IsPositive() {
+		return types.IncentiveRecord{}, types.NonPositiveIncentiveAmountError{PoolId: poolId, IncentiveAmount: incentiveAmount.ToDec()}
+	}
+
+	// Ensure start time is >= current blocktime
+	if startTime.Before(ctx.BlockTime()) {
+		return types.IncentiveRecord{}, types.StartTimeTooEarlyError{PoolId: poolId, CurrentBlockTime: ctx.BlockTime(), StartTime: startTime}
+	}
+
+	// Ensure emission rate is nonzero and nonnegative
+	if !emissionRate.IsPositive() {
+		return types.IncentiveRecord{}, types.NonPositiveEmissionRateError{PoolId: poolId, EmissionRate: emissionRate}
+	}
+
+	// Ensure min uptime is one of the supported periods
+	validUptime := false
+	for _, supportedUptime := range types.SupportedUptimes {
+		if minUptime == supportedUptime {
+			validUptime = true
+		}
+	}
+	if !validUptime {
+		return types.IncentiveRecord{}, types.InvalidMinUptimeError{PoolId: poolId, MinUptime: minUptime, SupportedUptimes: types.SupportedUptimes}
+	}
+
+	// Ensure sender has balance for incentive denom
+	incentiveCoin := sdk.NewCoin(incentiveDenom, incentiveAmount)
```

Comment:
related to earlier comments - I think having it as `Coin` from the start makes more sense to me as well